rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,59 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team:
+shepherd-leader:
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/) (#community:nixos.org) as our official
+community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space (#community:nixos.org) that has already been created which you
+can join by visting the [Invite
+link](https://app.element.io/#/room/#community:nixos.org).
```

Comment:
Notable difference: it does require signing up now in contrast to the ephemeral IRC identity you had before that.

Reply:
This might raise the bar for contributing - e.g. Rust project uses a chat which also requires sign-up (they allow to sign up with social media or Git forge accounts though via OAuth) and that makes me a little bit uncomfortable.

Reply:
There is an option **Who can access this room?** in the Element room settings, with one of the three options being **Anyone who knows the room's link, including guests**.

Currently, guests are allowed in #chat:nixos.org, but not #nix:nixos.org. Perhaps allowing guests in a few central rooms (or all?), and then hosting a neat guest widget without all of the registration flow might suffice here?
The chat widget might automatically join a specific room, or drop the user in the spaces view whenever clients mature.

Reply:
Do we actually have software that does this already? I have a dedicated HS for "guests" that is supposed to provide ephemeral identities but right now Element requires you to sign-up to participate in a conversation. I believe the feature we are looking for is called ILAG and apparently that might make a comeback at some point: https://github.com/vector-im/element-web/issues/9264#issuecomment-818587925

Reply:
Regarding the need for account creation, matrix is actually somewhere in-between of IRC (no account needed) and rocket chat/ mattermost/ zulip, where you need to create a new account for each community.
The beauty of federated open standards is that you need an account, but it's only a single one on an (almost) arbitrary server that can be used for all communities. The more communities are adopting that system, the less important is the assumption of people having an account. Thus IMHO that is an acceptable compromise.

Bringing back guest access would be neat anyways, although that might include a spam problem.